### PR TITLE
Compute global setup to avoid duplicate info

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
should do the same for extra
to build a real packages:
- cp tools/packages/taipy-<pkg>/* .
- python setup.py build_py
- python -m build

Taipy depêndencies version will have to be set when building the packages.

### CODE REVIEW

**Review by FabienLelaquais — DISMISSED** [review]
Far too many things that I don't get.
Whatever works.
Well done!

File: setup.py [hunk 1]
```diff
@@ -21,51 +21,56 @@
...
+# ]
 
 
 def get_requirements():
-    # TODO get requirements from the different setups in tools/packages (removing taipy packages)
-    return requirements
+    # get requirements from the different setups in tools/packages (removing taipy packages)
+    reqs = []
+    for pkg in (root_folder / "tools" / "packages").iterdir():
+        print(f"get reqs from {pkg}")
```
Comment (florian-vuillemot) [inline setup.py]:
```suggestion
```

File: setup.py [hunk 2]
```diff
@@ -21,51 +21,56 @@
...
+#     "kthread>=0.2.3,<0.3",
+#     "gitignore-parser>=0.1,<0.2",
+#     "simple-websocket>=0.10.1,<1.0",
+#     "twisted>=23.8.0,<24.0",
+#     "flask-restful>=0.3.9,<0.4",
+#     "passlib>=1.7.4,<1.8",
+#     "marshmallow>=3.20.1,<3.30",
+#     "apispec[yaml]>=6.3,<7.0",
+#     "apispec-webframeworks>=0.5.2,<0.6",
+# ]
```
Comment (florian-vuillemot) [inline setup.py]: To keep?

File: setup.py (hunk 1 above)
Comment (FabienLelaquais) [reply setup.py]: How about `print(f"Reading requirements from package '{pkg}'")`

File: setup.py (hunk 2 above)
Comment (FredLL-Avaiga) [reply setup.py]: of course not

File: setup.py (hunk 1 above)
Comment (FredLL-Avaiga) [reply setup.py]: the print is not seen in the logs => remove

**Review by FabienLelaquais — APPROVED** [review]
👌

File: tools/packages/taipy-rest/setup.requirements.txt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+flask>=3.0.0,<3.1
+flask-restful>=0.3.9,<0.4
+passlib>=1.7.4,<1.8
+marshmallow>=3.20.1,<3.30
+apispec[yaml]>=6.3,<7.0
+apispec-webframeworks>=0.5.2,<0.6
+taipy-core
```
Comment (florian-vuillemot) [inline tools/packages/taipy-rest/setup.requirements.txt:7]: No version?

**Review by florian-vuillemot — APPROVED** [review]
Great! But how are we managing taipy dependencies? I mean, setting the taipy-core/gui/etc version in the taipy package?

File: tools/packages/taipy-rest/setup.requirements.txt (hunk 3 above)
Comment (FredLL-Avaiga) [reply tools/packages/taipy-rest/setup.requirements.txt:7]: that's a subject for a next PR

**Review by florian-vuillemot — APPROVED** [review]

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**